File: src/main/java/org/extrasapache/myfaces/codi/examples/ebean/support/data/SpreadSheetController.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.extrasapache.myfaces.codi.examples.ebean.support.data;

import javax.enterprise.context.Dependent;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

/**
 * @author Werner Punz (latest modification by $Author$)
 * @version $Revision$ $Date$
 *          <p/>
 *          controller which keeps track of the spreadsheeting state
 */

@Dependent
public class SpreadSheetController implements Serializable {
    Set<Comparable> _editable = new HashSet<Comparable>();

    public SpreadSheetController() {

    }

    public void clear() {
        _editable.clear();
    }

    public void enableEdit(Identifyable instance) {
        _editable.add(instance.getId());
    }

    public void disableEdit(Identifyable instance) {
        _editable.remove(instance.getId());
    }

    public Boolean isEditable(Identifyable instance) {
        return _editable.contains(instance.getId());
    }

    public boolean isEmpty() {
        return _editable.isEmpty();
    }
}

File: src/main/scala/org/extrasapache/myfaces/codi/examples/ebean/support/data/SpreadSheetController.scala
package org.extrasapache.myfaces.codi.examples.ebean.support.data

import collection.mutable.HashSet
import javax.inject.Named
import javax.enterprise.context.Dependent

/**
 *
 * @author Werner Punz (latest modification by $Author$)
 * @version $Revision$ $Date$
 */

@Named
@Dependent
@serializable
class SpreadSheetController {
  private[this] var editable = new HashSet[java.lang.Long]

  def enableEdit(instance: StdEntity) = editable.add(instance.id)

  def disableEdit(instance: StdEntity) = editable.remove(instance.id)

  def isEditable(instance: StdEntity): Boolean = editable.contains(instance.id)

  def isEmpty: Boolean = editable.isEmpty

  def clear = editable.clear

}